moved/src/move_execution/tests/utils.rs

```diff
@@ -0,0 +1,631 @@
+            self.state
+                .resolver()
+                .get_module(&module_id)
+                .unwrap()
+                .is_some(),
+            "Code should be deployed"
+        );
+        module_id
+    }
+
+    /// Executes a Move script with arguments
+    ///
+    /// # Arguments
+    /// * `script_name` - Name of the script to execute
+    /// * `local_deps` - Local module dependencies needed by the script
+    /// * `args` - Arguments to pass to the script
+    pub fn run_script(
+        &mut self,
+        script_name: &str,
+        local_deps: &[&str],
+        args: Vec<TransactionArgument>,
+    ) -> Vec<Log<LogData>> {
+        let script_bytes = self.compile_script(script_name, local_deps, args);
+        let (tx_hash, tx) = create_transaction(&mut self.signer, TxKind::Create, script_bytes);
+        let transaction = TestTransaction::new(tx, tx_hash);
+        let outcome = self.execute_tx(&transaction).unwrap();
+        self.state.apply(outcome.changes).unwrap();
+        // Script transaction should succeed
+        outcome.vm_outcome.unwrap();
+        outcome.logs
+    }
+
+    /// Transfers ETH to a specified address with L1 cost considerations
+    ///
+    /// # Arguments
+    /// * `to` - Destination address for the transfer
+    /// * `amount` - Amount of ETH to transfer
+    /// * `l1_cost` - L1 cost associated with the transaction
+    /// * `expect_error` - Whether the transfer is expected to fail
+    pub fn transfer(&mut self, to: Address, amount: u64, l1_cost: u64, expect_error: bool) {
+        let (tx_hash, tx) = create_transaction_with_value(
+            &mut self.signer,
+            TxKind::Call(to),
+            Vec::new(),
+            U256::from(amount),
+        );
+
+        // Default base token is ETH token in address 0x1
+        let treasury_address = AccountAddress::ONE;
+        let base_token = MovedBaseTokenAccounts::new(treasury_address);
+        let mut transaction = TestTransaction::new(tx, tx_hash);
+        transaction.with_cost(l1_cost, base_token);
+        let outcome = self.execute_tx(&transaction).unwrap();
+        if expect_error {
+            outcome.vm_outcome.unwrap_err();
+        } else {
+            outcome.vm_outcome.unwrap();
+        }
+        self.state.apply(outcome.changes).unwrap();
+
+        let treasury_balance = self.get_balance(treasury_address.to_eth_address());
+        assert_eq!(treasury_balance, l1_cost);
+    }
+
+    /// Executes a Move entry function with the given arguments
+    ///
+    /// This is the recommended way to call Move functions in tests as it handles
+    /// argument serialization and transaction creation.
+    ///
+    /// # Arguments
+    /// * `module_id` - The ModuleId containing the function to execute
+    /// * `function` - Name of the function to call
+    /// * `args` - Vector of Move values to pass as arguments
+    pub fn execute(&mut self, module_id: &ModuleId, function: &str, args: Vec<&MoveValue>) {
```

Comment:
Nit: there's no reason for `args` to be `Vec`. And type implementing `IntoIterator<Item=&MoveValue>` would work (for example an array instead of `Vec`). Having the implementation be generic could avoid some unnecessary allocations.